cnap/core/infereng.py

```diff
@@ -0,0 +1,332 @@
+            self._id = uuid.uuid1()
+        return str(self._id)
+
+    @id.setter
+    def id(self, new_str: str) -> None:
+        """
+        Set UUID for Infer Engine Info
+
+        Args:
+            new_str: UUID for Infer Engine Info
+
+        Returns:
+            None
+        """
+        self._id = uuid.UUID(new_str)
+
+    @property
+    def input_size(self) -> Tuple[int, int]:
+        """
+        Input size tuple required by infer model
+
+        Returns:
+            Input size tuple required by infer model
+        """
+        return self._input_size
+
+    @property
+    def device(self) -> str:
+        """
+        Device type for infer engine
+
+        Returns:
+            Device type for infer engine
+        """
+        return self._device
+
+    @property
+    def model_id(self) -> str:
+        """
+        UUID for infer model
+
+        Returns:
+            UUID for infer model
+        """
+        return self._model_id
+
+    @property
+    def queue_topic(self) -> str:
+        """
+        Queue topic for infer engine
+
+        Returns:
+            Queue topic for infer engine
+        """
+        if self._queue_topic is None:
+            self._queue_topic = f"origin-{self.model_id}-{self.device}"
+        return self._queue_topic
+
+    @queue_topic.setter
+    def queue_topic(self, new_queue_topic: str) -> None:
+        """
+        Set queue topic for infer engine
+
+        Args:
+            new_queue_topic: Queue topic for infer engine
+
+        Returns:
+            None
+        """
+        self._queue_topic = new_queue_topic
+
+    def __iter__(self) -> Iterator[Tuple[str, Any]]:
+        yield 'id', self.id
+        yield 'device', self.device
+        yield 'model_id', self.model_id
+
+class InferEngineManager:
+    """
+    To manage inference engines by registering them and searching for available engines.
+    """
+
+    _instance = None
+
+    def __init__(self, db: RuntimeDatabaseBase) -> None:
+        """
+        Initialize the InferEngineManager
+
+        Args:
+            db: RuntimeDatabaseBase object
+
+        Returns:
+            None
+        """
+        self._db = db
+
+    def search_engine(self, framework: str, target: str, device: str,
+                      model_name: str, model_version: str) -> Optional[InferenceInfo]:
+        """
+        Search the engine according to the framework and target requested.
+        return the InferenceInfo include the UUID of infer engine
+
+        Args:
+            framework: Framework of the model
+            target: Target of the model
+            device: Device type for infer engine
+            model_name: Name of the model
+            model_version: Version of the model
+
+        Returns:
+            InferenceInfo object if engine is found, None otherwise
+        """
+        # Get all available engines from database
+        engine_dicts = self._db.get_all_table_objects_dict(INFER_ENGINE_TABLE)
```

Comment:
Should we catch the error raised by `rtdb` here?

Reply:
Fixed.